src/basic/macro.h

```diff
@@ -30,15 +30,6 @@
 #define _function_no_sanitize_float_cast_overflow_
 #endif
 
-#if HAVE_WSTRINGOP_TRUNCATION
```

Comment:
This is the only occurrence of the check, hence please drop it from meson.build too

Reply:
`stringop-truncation` was introduced in gcc8, and our baseline is gcc8.4. Hence, we can drop `HAVE_WSTRINGOP_TRUNCATION`.